.github/workflows/build.yml

```diff
@@ -95,10 +99,79 @@ jobs:
       run: find tools/ -type f -iname "*.sh" -exec chmod +x {} \;
 
     - name: Setup python
-      uses: actions/setup-python@v4
+      uses: actions/setup-python@v5
       with:
         python-version: ${{ matrix.python_version }}
 
+    - name: List available Xcode versions
+      run: ls /Applications | grep Xcode
+
+    - name: Set up Xcode version
```

Comment:
Is this necessary now because of changes to Github's MacOS runners?

Reply:
Yeah. I was seeing a build fail on the default 15.4 xcode on macos-14